libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -533,10 +506,11 @@ static void serializePublishCommon( const MQTTPublishInfo_t * pPublishInfo,
         LogDebugWithArgs( "Copying PUBLISH payload of length =%lu to buffer",
                           pPublishInfo->payloadLength );
 
-        /* This memcpy intentionally copies bytes from a void * buffer into
-         * a uint8_t * buffer. */
-        /* coverity[misra_c_2012_rule_21_15_violation] */
-        ( void ) memcpy( pIndex, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+        /* Typecast const void * typed payload buffer to const uint8_t *.
+         * This is to use same type buffers in memcpy. */
+        pPayloadBuffer = ( const uint8_t * ) pPublishInfo->pPayload;
```

Comment:
Isn't casting from void * to anything another MISRA violation? So we'll have to suppress a violation anyway

Reply:
No. This cast didn't throw any violation when using our config

Reply:
That's likely due to this: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/scripts/coverity_misra.config#L35

Should we consider re-enabling it for this branch?

Reply:
That brings up a good point, I think we should evaluate all the deviations that we have added in the config for the development branch.
@abhidixi11 What do you think?